sprocs/assessment_instances_insert.sql

```diff
@@ -75,25 +75,36 @@ BEGIN
 
     -- ######################################################################
     -- do the actual insert
+    -- ON CONFLICT: make sure there is no new instance created by student's teammate simultaneously
+    -- ON CONFLICT will not update anything but for returning the id
 
     INSERT INTO assessment_instances
             ( auth_user_id, assessment_id, user_id,     group_id, mode, auto_close, date_limit, number)
-    VALUES  (authn_user_id, assessment_id, 
+    VALUES  (authn_user_id, assessment_id,
             CASE WHEN group_work THEN NULL      ELSE user_id END,
             CASE WHEN group_work THEN tmp_group_id ELSE NULL END, mode, auto_close, date_limit, number)
+    ON CONFLICT (assessment_id, group_id, number) DO UPDATE
+    SET assessment_id = excluded.assessment_id
     RETURNING id
     INTO assessment_instance_id;
 
+    -- ######################################################################
+    -- A null assessment_instance_id means we triggered ON CONFLICT and DO NOTHING
```

Comment:
There's no `DO NOTHING` clause above.
```suggestion
    -- A null assessment_instance_id means we triggered ON CONFLICT
```